pkg/apis/eventing/v1alpha1/trigger_lifecycle.go

```diff
@@ -45,11 +45,21 @@ func (ts *TriggerStatus) GetCondition(t apis.ConditionType) *apis.Condition {
 	return triggerCondSet.Manage(ts).GetCondition(t)
 }
 
+// GetCondition returns the condition currently associated with the given ConditionType happy, or nil.
+func (ts *TriggerStatus) GetHappyCondition() *apis.Condition {
+	return triggerCondSet.Manage(ts).GetHappyCondition()
+}
+
 // IsReady returns true if the resource is ready overall.
 func (ts *TriggerStatus) IsReady() bool {
 	return triggerCondSet.Manage(ts).IsHappy()
 }
 
+// IsReady returns true if the resource is unknown overall.
+func (ts *TriggerStatus) IsUnknown() bool {
+	return triggerCondSet.Manage(ts).IsUnknown()
```

Comment:
just use  `triggerCondSet.Manage(bs).GetCondition(TriggerConditionReady)` ?

Reply:
Is return triggerCondSet.Manage(ts).IsUnknown() more clear here? Both IsReady() and IsUnknown() follows the same style? If we use `triggerCondSet.Manage(bs).GetCondition(TriggerConditionReady).IsUnknown()` for `IsUnknwon' and use `triggerCondSet.Manage(ts).IsHappy()` for `IsReady`, is it quite confusing since we have two different styles when they operates on the same condition of same resource? I prefer adding `IsUnkown` to `ConditionManager` Interface if it doesn't break other knative resources.